libs/ai-endpoints/langchain_nvidia_ai_endpoints/chat_models.py

```
@@ -822,6 +814,59 @@ def _get_payload(
             else:
                 raise ValueError(f"Unknown message received: {msg} of type {type(msg)}")
 
+        # Handle thinking mode via parameters or prefix to system message
+        thinking_mode = kwargs.pop("thinking_mode", None)
+        if thinking_mode is not None and self._client.model:
+            # Check if model uses param-based thinking
+            thinking_params = (
+                self._client.model.thinking_param_enable
+                if thinking_mode
+                else self._client.model.thinking_param_disable
+            )
+
+            if thinking_params:
+                # Param-based thinking: merge parameters into kwargs
+                def deep_merge(base: dict, update: dict) -> dict:
+                    """Deep merge update dict into base dict."""
+                    result = base.copy()
+                    for key, value in update.items():
+                        if (
+                            key in result
+                            and isinstance(result[key], dict)
+                            and isinstance(value, dict)
+                        ):
+                            result[key] = deep_merge(result[key], value)
+                        else:
+                            result[key] = value
+                    return result
+
+                kwargs = deep_merge(kwargs, thinking_params)
+            else:
```

Can a model be misconfigured with both tag-based and param-based thinking at the same time mistakenly by users when they use their own configs? The code silently uses param-based and ignores tag-based, should we add some kind of warning when both is set?